javascript/javascript3/week2/VisualPromise/main.js

```diff
@@ -0,0 +1,60 @@
+const redBox = document.querySelector("ul.marks li:nth-child(1)");
+const blueBox = document.querySelector("ul.marks li:nth-child(2)");
+const greenBox = document.querySelector("ul.marks li:nth-child(3)");
+
+const redTargetX = 20 - parseInt(redBox.style.left);
+const redTargetY = 300 - parseInt(redBox.style.top);
+const blueTargetX = 400 - parseInt(blueBox.style.left);
+const blueTargetY = 300 - parseInt(blueBox.style.top);
+const greenTargetX = 400 - parseInt(greenBox.style.left);
+const greenTargetY = 20 - parseInt(greenBox.style.top);
+
+function translateOneByOne() {
+  moveElement(redBox, {
+    x: redTargetX,
+    y: redTargetY,
+  })
+    .then(() => console.log("Red Box moved"))
+    .then(() =>
+      moveElement(blueBox, {
+        x: blueTargetX,
+        y: blueTargetY,
+      })
+    )
+    .then(() => console.log("Blue Box moved"))
+    .then(() =>
+      moveElement(greenBox, {
+        x: greenTargetX,
+        y: greenTargetY,
+      })
+    )
+    .then(() => console.log("Green Box moved"));
+}
+
+translateOneByOne();
+
+//Comment out whichever you'd like to use
+
+// function translateAllAtOnce() {
+//   try {
+//     const redBoxPromise = moveElement(redBox, {
+//       x: redTargetX,
+//       y: redTargetY,
+//     });
+//     const blueBoxPromise = moveElement(blueBox, {
+//       x: blueTargetX,
+//       y: blueTargetY,
+//     });
+//     const greenBoxPromise = moveElement(greenBox, {
+//       x: greenTargetX,
+//       y: greenTargetY,
+//     });
```

Comment:
You could simplify this a little by creating the array here, like this:
```js
const movements = [
  moveElement(redBox, redTarget),
  moveElement(blueBox, blueTarget),
  moveElement(greenBox, greenTarget),
];

Promise.all(movements).then(...)
```

I usually find that if a variable includes the word promise but isn't some promise function helper then it usually means we're missing to name some other concept, in this case the movements we're waiting for! 

Also note that try/catch doesn't work as you'd expect with Promises. It will only catch errors that happen when creating the Promise, not when the Promise runs into issues while running. Instead, you'd use the `.catch` handler that can be added after any Promise chain to handle any errors that happen inside the Promise chain.